soap/permissions.py

```
@@ -0,0 +1,25 @@
+
+from typing import Any
+from rest_framework.permissions import BasePermission
+
+
+class IsDoctor(BasePermission):
+    """
+    دسترسی فقط برای پزشکان
+    """
+    message = 'دسترسی فقط برای پزشکان'
+
+    def has_permission(self, request: Any, view: Any) -> bool:
```

Ruff (0.12.2)

Dynamically typed expressions (typing.Any) are disallowed in `request`

(ANN401)